argo-cd-apps/base/enterprise-contract.yaml

```diff
@@ -1,77 +1,52 @@
-    targetRevision: "143b8d71fc861d7c7ac93a15d0fbd33654b75908"
-
-    helm:
-      valueFiles:
-        - values.yaml
-      values: |-
-        server:
-          strategy:
-            type: Recreate
-          ingress:
-            hostname: rekor-server.enterprise-contract-service.svc
-            annotations:
-              route.openshift.io/termination: "edge"
-        mysql:
-          auth:
-            password: "password"
-
-  destination:
-    namespace: enterprise-contract-service
-    server: https://kubernetes.default.svc
-
-  syncPolicy:
-    # Comment this out if you want to manually trigger deployments (using the
-    # Argo CD Web UI or Argo CD CLI), rather than automatically deploying on
-    # every new Git commit to your directory.
-    automated:
-      prune: true
-      selfHeal: true
-
-    syncOptions:
-      - CreateNamespace=true
-
-    retry:
-      limit: 50 # number of failed sync attempt retries; unlimited number of attempts if less than 0
-      backoff:
-        duration: 15s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")
 ---
 apiVersion: argoproj.io/v1alpha1
-kind: Application
+kind: ApplicationSet
 metadata:
   name: enterprise-contract
-
+  namespace: openshift-gitops
 spec:
-  project: default
-
-  source:
-    path: components/enterprise-contract
-    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
-    targetRevision: main
-
-  destination:
-    namespace: enterprise-contract-service
-    server: https://kubernetes.default.svc
-
+  generators:
+    - list:
+        elements:
+          # Determines which Rekor (Tekton Chains transparency log) variant will
+          # be used, can be:
+          #  * off      - transparency off
+          #  * local    - transparency on, with an in-cluster Rekor instance
+          #  * sigstore - transparency on, with Sigstore public Rekor instance
+          - rekor: off
   syncPolicy:
-
-    # Comment this out if you want to manually trigger deployments (using the
-    # Argo CD Web UI or Argo CD CLI), rather than automatically deploying on
-    # every new Git commit to your directory.
-    automated:
-      prune: true
-      selfHeal: true
-
-    retry:
-      limit: 50 # number of failed sync attempt retries; unlimited number of attempts if less than 0
-      backoff:
-        duration: 15s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")
+    # Allows us to delete the "rekor" Application if "rekor" above is not "local"
+    preserveResourcesOnDeletion: false
+  template:
+    metadata:
+      name: enterprise-contract
+
+    spec:
+      project: default
+      source:
+        # Uses the overlay specified by "rekor" generator property above, this
+        # will setup in-cluster Rekor (if set to "local"), and change the
+        # chains-config ConfigMap based on that setting
+        path: components/enterprise-contract/rekor-{{rekor}}
```

Comment:
I'm trying to understand this (so bear with me!).

Are you using an ApplicationSet just as a means to select a different path? And if so, to use a different mode of Rekor, we'd have to change the value in the generators list above? Wouldn't it be easier to just use a regular Application and document that to use a different mode, you must change the path to a different one?

Reply:
Regular Applications don’t garbage collect, so switching from “rekor-local” to “off” would keep the Rekor running